### DISCUSSION TITLE
Dynamic row count on Datatable component?

### DISCUSSION THREAD

**Discussion by dominic-henley** [shadcn-ui/ui#4060] [Q&A]
Hey everyone, just started using shadcn and loving it so far. I'm looking to implement a datatable and I've been following the short tutorial on the documentation [page](https://ui.shadcn.com/docs/components/data-table#basic-table). The pagination works well however it defaults to 10 rows. This leaves a big gap between the pagination controls and the datatable itself. I've attached a screenshot of the datatable in question.
![image](https://github.com/shadcn-ui/ui/assets/80395298/ab381312-1d21-447a-a95a-34dbf91e7306)

I was wondering if there's a way to determine the number of rows per page dynamically?

I've thought about setting the number of rows manually by taking the height of the table assuming it's taking up the whole space and dividing by the number of rows + header but I'm not sure if the height of the rows will be guaranteed to be the same everywhere.

Anyone have any suggestions? Thanks in advance!

**Comment by ezeparziale** [comment]
The pagination component always remains after the last row, whether you have 10, 20, or 50 rows. I don't know what your code looks like.
You can check this example: https://github.com/shadcn-ui/ui/blob/main/apps/www/app/(app)/examples/tasks/components/data-table.tsx
